### ISSUE TITLE
Welcome, Outreachy people!

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#3679]
Hello folks from Outreachy - I realized we don't have a main place for you all to ask questions and get support, so I wanted to say WELCOME and create this issue as a discussion zone. Please leave a comment to say hello, in part so we know who's interested in the program! @publiclab/reviewers @publiclab/mentors please keep an eye open for opportunities to welcome and support our Outreachy applicants! 👍 🎉 

## First contribution

It's so great to see that several of you have made your first contribution. 🙌 This should really help as all our contributions follow this workflow. If you haven't yet, please see this page to get started:

https://code.publiclab.org

## What next?

If you're looking for a next project, I might recommend one which starts to involve multiple files in the application code, so you can start to get a better understanding of the application's workings. Are you familiar with how a Rails application works, in terms of how the model accesses the database, the controller performs the logic, and the views display the actual web pages? If not, we can help you start to pick apart how our system works one piece at a time. 

You can find a lot of relatively small but not completely documented issues here:

https://github.com/publiclab/plots2/labels/fto-candidate

If you choose one, please leave a comment and we can do our best to help find any missing information or help guide you through the next steps. Don't hesitate to ask questions!

Larger collections of tasks can often be found under the `break-me-up` label, which you can go through to find smaller projects to "break off" and solve:

https://github.com/publiclab/plots2/labels/break-me-up

## Supporting tasks

Another area you might be interested in contributing is in supporting other new people to the project. This is an often unrecognized part of a healthy collaborative project, so we've made these issues to encourage people to take part in this crucial activity. We'd love your help, and as a newcomer yourself, you know exactly what it feels like to be new, so you know how important it is to be welcoming and supportive!

https://github.com/publiclab/plots2/labels/support

Thanks and we're happy to answer questions!!

****

### Update:

I've posted a call for proposals here, where I'd like to ask each of you to open a draft proposal and get some comments on it! Thank you!!

https://publiclab.org/notes/warren/10-23-2018/call-for-outreachy-2018-19-proposals

**Comment by jywarren** [comment]
@quinn-codes-synthesis @faithngetich and more who've reached out -- thanks!

**Comment by sagarpreet-chadha** [comment]
@cesswairimu @Kanikashrivastav also 😄 🎉 !

Please note to record your in-progress contributions as well at : 
https://www.outreachy.org/december-2018-march-2019-outreachy-internships/communities/public-lab/extend-community-collaboration-statistics-and-visu/contributions/

**Comment by quinn-codes-synthesis** [comment]
Hellooooooooooo! 

![Image of Ariel waving](http://www.lovethisgif.com/uploaded_images/87329-Ariel-Waves-Hello-Excited-In-The-Little-Mermaid-Gif.gif)
(_image description: gif of Ariel from The Little Mermaid waving enthusiastically_)

A quick intro:

I'm Quinn, I live in the US, and my pronouns are they/them/theirs. I've dabbled in HTML, CSS, and general design/layout/illustration stuff since I was young, but it wasn't until this past May that I started dedicated study of Javascript, more advanced HTML/CSS, and recently Ruby. My intent is to make programming a career, because I think the industry will benefit from people like me who want to find technological solutions to human problems -- and have expertise in both. 

My work and educational history is in psychology and behavioral health, but you might be surprised at the parallels between that and programming. Helping someone through a mental health crisis isn't all that different from debugging a computer program. There are a few tried and true methods that are known to help in most situations, so we try those first. In psychology, this might be a course of Cognitive Behavioral Therapy, and in coding it might be something like making sure there are no typos or errors in your syntax. If you try a few basic things and they don't work, you have to start getting creative with your solutions. Maybe you have to spend a few hours, days, weeks working through resolving a patient's trauma or reading through pages and pages of Stack Overflow. But there is _always_ a solution, even if it takes a while to find it!

Aside from coding and psych, I'm also a _huge_ sci-fi nerd. My username is a sci-fi reference (to a great little video game series called Mass Effect). When learning programming gets tough, sci-fi is my inspiration to keep going. I think of all the technologies past generations dreamed of that are similar to tech we have today -- like the tablets used in Star Trek, for instance -- because someone had the diligence to figure out how to make it a reality. Just gotta get together with some engineering wizards and make it happen, one line of code at a time. :)

**Comment by SidharthBansal** [comment]
Great to see you guys here.
We will love to have you on Public lab.

**Comment by jywarren** [comment]
Hi!!! Welcome!!!

**Comment by Codeama** [comment]
Hi,

How do I make a first contribution seeing all the first-timer issues are either reserved or get taken up so fast?

**Comment by amychan331** [comment]
Hi, this is Amy from San Francisco! I immigrated from Hong Kong when I was young. Having lived in 2 metropolitan and coming from a low-income family, I have always been interested in the environment - both in term of natural and artificial materials - and their combined affect on local, vulnerable communities. As a result, I studied architecture, interned in a community design firm, and then became a LEED Green Associate right after I graduate.
While building my online portfolio, I fell in love with coding. I taught myself to programming with online tutorials, night classes at community college, and a lot of determination. I don't know Ruby for now, but I have always quickly picked up whatever tools I need when I am passionate about a subject, and environmental advocacy has been a passion since I took my first AP Environment Science class. 
I hope to combine my interest in programming, design, and environment in my future work. Needless to say, I was enthusiastic when I saw Public Lab and realized that that an environmental-focused organization joined Outreachy. I know it's already mid-October, but I hope that I am not too late!

**Comment by aSquare14** [comment]
Hi @amychan331  ! No you're not too late ! Welcome to publiclabs.  Let us know if you need help picking out a good first issue !

**Comment by quinn-codes-synthesis** [comment]
@Codeama -- maybe we could both help each other on this. I'm now on the step where I should setup a welcoming issue for someone else. I could create an issue and reserve it for you, like @jywarren did for me (thanks again)! I just haven't had a chance in the last few days to look over the codebase and find one. I've got the next few hours mostly free, so I'll devote that time to this and tag you again as soon as I've got one.

**Comment by Codeama** [comment]
@quinn-codes-synthesis:  Thanks, Quinn. I'll be on the lookout for an issue!

**Comment by quinn-codes-synthesis** [comment]
https://github.com/publiclab/plots2/issues/3701 -- Here you go!

**Comment by mardelvalle** [comment]
Hi all! My name is Maria Cristina. I am happy to meet you all. I am a first-generation American. My background of the study was actually originally environmental science and anthropology, however, I am now focusing on coding but haven't lost my interest in those subjects 

I have primarily worked with HTML, CSS, and JavaScript but am always eager to learn new skills! Besides coding, you can find me drawing, knitting and going to the movies.

**Comment by jywarren** [comment]
Hello! Just checking back in here. Welcome, @mardelvalle -- very glad to meet you too! We are very happy to get you set up and started. Have you found a first-timers issue yet or can we help you find one to work on?

**Comment by jywarren** [comment]
And it's great to hear of your background -- some of the founders of Public Lab were anthropologists and environmental scientists, along with teachers, designers, artists, and programmers. So, welcome!

**Comment by mardelvalle** [comment]
Hi @jywarren nice to meet you! I figured that there were people with environmental background on the project but I didn't realize that there were people with anthropology as well!

I haven't been able to find an issue yet. It looks like there is activity on all of the first-timers issues as far as I can tell.

**Comment by jywarren** [comment]
And welcome to @amychan331 -- I studied architecture myself, actually! 😄 

@Codeama welcome! I see you and @quinn-codes-synthesis have worked through and solved an issue and we're so grateful.

**Comment by jywarren** [comment]
I've made a new one for you! Mentioning you in it now so you'll get a notification.

**Comment by mardelvalle** [comment]
Awesome. Thank you!

**Comment by jywarren** [comment]
Folks are starting to ask about proposals. Im about to board a plane but I want to put out this link with some examples of proposals from our summer of code programs. I need to check the exact requirements in Outreachy but these have some of the kind of information tthat will be helpful in framing out a proposal:

https://publiclab.org/notes/warren/02-28-2017/call-for-proposals 

I hope this is helpful! Thanks and more soon!

**Comment by jywarren** [comment]
Ah, and here's the one for this last year's program! The last link was for 2017. 

https://publiclab.org/notes/warren/02-15-2018/call-for-summer-of-code-2018-proposals

**Comment by faithngetich** [comment]
Thank-you @jywarren

**Comment by SidharthBansal** [comment]
Hi folks, please start writing your proposals so that the mentors can suggest changes to it. While jeff is travelling other people will love to have a look at your proposals.

**Comment by SidharthBansal** [comment]
A few tips for making awesome proposals:
* Your information so that we can know you better.
* Include weekly timeline: Explain what you will be doing in which week. Divide the Outreachy period into months in bigger tasks. Then into weeks into smaller tasks.
* Include demos/images/gifs to demonstrate your idea
* Include Mock Ups, Flow Charts, Data Flow Diagrams, etc to explain the workflow in simple language. So that people in logistic department can review your work and suggest changes.
* PRs merged & Issues reported: This is the most crucial aspect of any application. We want people to be well versed with our system. Please mention all the issues reported, pr merged, suggestions etc in the proposal.
* Start early and have many revisions of the proposal to increase your chances.

I have also not checked the official documentation till yet. Was busy due to some work. These tips are general tips. You need to submit the proposals at publiclab.org first. After all the reviews you can submit it on the official website.

Are you tensed? No need to worry!!! :smile: We all are there to help you at each stage of proposing the proposal to the manufacturing of the end product. So just be happy and code.

Best of luck folks. Hope we will have great time this year with you all.

**Comment by Kanikashrivastav** [comment]
Hello All. I just noticed, I haven't introduced myself to the community.  My name is Kanika Shrivastav. I live in Delhi, India. My interest in coding actually started one year ago. I was quite fascinated by web and android application and the fact the you can build apps to solve common problems. I started with HTML, CSS and Python. It was very hard to learn stuff on my own. I remember, I couldn't make a simple webpage. But I tried very hard and looked up various you-tube videos and finally I could make a web page. I was also introduced to an organization called Navgurukul which helps student like me to learn coding. So I was looking to decide which framework to learn next to build full stack web application. I finally decided to learn rails for a couple of reasons. As I read more blogs about this, building application in rails is very fast. and writing database query is not that complex. So I decided to learn Rails. I have only build a small CRUD app (not using the rails command). I'm still learning more stuff  in rails.  But I have learned so many things while solving the issues. Like using (I18n) in the views for providing multi-language support. Currently I'm learning about catching and much more. This is the reason I love open source as I got to learn many new things which I couldn't have imagined learning myself.  I'm really excited to do more contribution to this community and have a good learning experience.

**Comment by jywarren** [comment]
Hi @Kanikashrivastav and @faithngetich ! 

That's awesome you have experience with i18n, @Kanikashrivastav -- we are in the midst of a big project to restructure ours, in #3262  -- and so cool you've made a CRUD app! I remember learning Rails myself on a very long (52 hour!) train ride across the USA, with a downloaded PDF manual and eating a lot of 🍜 noodles along the way 😄  It's very exciting to have a chance to work with you all as you build skills and discover your talents! 

I reviewed the Outreachy application process but it wasn't clear to me that you need to submit a project plan (at least not yet) so I think some of my suggestions above may be more than you need at this point in the application process. But I hope they're still helpful for understanding how we structure and plan out big projects and how you'll be supported in designing a timeline and schedule. 

If i've missed anything and you **do** have to complete a proposal at this point, please correct me! But I think that at this point if you have questions about the proposal we've posted here, please leave a comment on that issue:

#3498 (Planning for expanded community stats system)

Thanks and sorry to have misunderstood the process a bit -- this is our first year with Outreachy! 🎉

**Comment by SidharthBansal** [comment]
Folks are requested to brainstorm the ideas related to the project together on https://github.com/publiclab/plots2/issues/3726 and on their posts on Public labs.
We really appreciate collaboration at Public labs

**Comment by Codeama** [comment]
@jywarren I agree there was no indication of a project plan on the Outreachy application. So this is very new information to me. And just to check, @SidharthBansal,  is the project proposal _now_ part of the selection criteria? This would help me in deciding whether or not to complete my Outreachy application as there would be no point submitting a half-finished one. But I am very much interested in contributing, whichever way this goes.

**Comment by sagarpreet-chadha** [comment]
Hi @Codeama !

So according to Outreachy Timeline [here](https://www.outreachy.org/apply/project-selection/) , the last date to record contributions and submit proposals is Oct 30 . So you still have many days .

If you still need more time --- i guess we have to decide the final selection of intern till **Nov 16**  , so you may submit partial proposal and your contributions before the deadline , i.e **Oct 30** on outreachy website and continue to improve / modify your proposal on[ publiclab.org]( publiclab.org) .


Hope this helps 😄 ! 

@jywarren @SidharthBansal

**Comment by SidharthBansal** [comment]
Bukolo yes this project proposal is the main criterion for selection in
Outreachy

**Comment by Codeama** [comment]
Thanks for the clarification, @SidharthBansal and @sagarpreet-chadha !

**Comment by jywarren** [comment]
Hi, everyone! I'm sorry there was some confusion -- since, unlike in our other Summer of Code programs, we have a single project we've submitted, we weren't sure we needed a call for proposals page. But although we'd love to see group discussion and collaboration around refining the proposals, it does seem helpful to ask each applicant to post a draft proposal (in addition to the one you'll send to Outreachy.org).

So I've posted a call for proposals here, where I'd like to ask each of you to open a draft proposal and get some comments on it! Thank you!!

https://publiclab.org/notes/warren/10-23-2018/call-for-outreachy-2018-19-proposals

**Comment by bansalsidharthcode** [comment]
University and college students who are applying are requested to provide information about their school calendar and class credits. We will ask for your current term and your next term. Please have that information before starting the process.

**Comment by SidharthBansal** [comment]
All the participants of Outreachy are requested to submit the application before October 30 at 4pm UTC on the main Outreachy website. Failure to do so will result in failure from the Outreachy program.

Applicants who create an application can edit it until the project application deadline. After your project's application deadline, applicants will NOT be able to create a new application. Applicants who have submitted an application cannot edit their application, but they will be able to continue recording and editing their contributions on the Outreachy website until the intern announcement deadline on Nov. 16, 2018 at 4pm UTC. Applicants who have not submitted an application will not be able to record or edit their contributions.

**Comment by jywarren** [comment]
Some folks have written in asking about a good follow-up project after their first contribution! It's a great question. There are always a lot marked "help-wanted" -- but for some where there may be a smaller, bite-sized one to take on, you could try looking at the label for "break-me-up" and finding one that hasn't yet been separated out. 

https://github.com/publiclab/plots2/labels/help-wanted

https://github.com/publiclab/plots2/labels/break-me-up

Please ask if you need help deciding if one is right for you!

**Comment by catherineattheodi** [comment]
Hey everyone,

I'm Catherine from Indonesia - hope I'm not too late to this discussion. Just graduated with a bachelors degree in combined social sciences (mainly Politics and Philosophy), and am undertaking my pre-masters gap year. I have work/internship experience in a few industries, most recently at a technology non-profit, where I worked on (amongst other things) refining the processing of open opportunity data for one of their sector programmes. From this, I gained an interest in figuring out where this data came from (since we worked with collated data given by partners, rather than collecting it ourselves through a system). I have some basic experience with Github through having to fix bugs and edit parts of the sectoral programme's website when I was interning (mostly html editing), and would love to learn more about Ruby on Rails through this project. 

I'm trying to start on a first contribution, but it seems many of the first contribution tasks have been reserved/claimed, so will see if I can slowly work on any of the other tasks available (which are not marked gci, as mentioned). :)

**Comment by jywarren** [comment]
Not too late, and welcome! Please keep an eye on the schedule though -- we
don't want folks to miss the deadlines.

**Comment by gauravano** [comment]
Hi everyone, if I guess the deadline is extended to Nov, 6 as per https://www.outreachy.org/apply/project-selection/#public-lab-extend-community-collaboration-statistics-and-visu . @jywarren please confirm. Thanks!

**Comment by Radhikadua123** [comment]
Hello everyone, I am Radhika Dua pursuing B.E( 4th year) from Panjab University, Chandigarh, India. This is my first patch in this project. I'm participating in the Outreachy program this year from where I got to know about this project. I read basics of both Ruby as well as Rails just few hours back. Since I had little experience with MVC based frameworks I was able to understand bits fast. I tried to find fto bugs but all of them seemed to be claimed. So, I went through github issues and tried to fix it.

I'll try to go through more issues and work on a better one. Though it would be great if someone can create one and assign it to me. I know I'm engaging in the community bit late, but I'll try to devote as much time as possible and try to catch up things fast.

While going through different issues on github, best thing I liked about this community is quite active and helpful people. I'm really excited to do more contributions to this community. 😄

Besides software engineering, I love to explore Deep Learning and Computer Vision areas. I am aiming to become full time researcher. Aside from hacking code, I love playing with cats. I enjoy swimming and cycling whenever I go back to my hometown.

@gauravano And it's great to know that project's deadline has been extended. :)

**Comment by sagarpreet-chadha** [comment]
Yes its confirmed...we have extended the deadline to Nov 6 , 4 PM UTC :hourglass_flowing_sand:  .

Hi @Radhikadua123 ! Welcome to PublicLab :smile:  .

**Comment by bansalsidharthcode** [comment]
Hi @Radhikadua123 <https://github.com/Radhikadua123> ! Welcome to PublicLab
😄 .

**Comment by jywarren** [comment]
Awesome, thanks Sagarpreet and Sidharth!

**Comment by narnt** [comment]
Hi, my name is Maria Nebesnaya, and I'm from Russia. I'm afraid I'm too late for apply because most of issues that I can resolve on my own already taken by someone else, but I'll try.

Last two years I was learning programming, mainly web programming, and I really fond of it. I have nice skills in javascript, html, css. I like to investigate complex tasks, learn something new everyday, work in a team, make something meaningful and amazing. Also, I like apply css or js effects and make web animations.

My first project was a visual novel. It was two years ago, so it was very far from nice quality, but I want to rewrite it someday.

I also learn python and a little learn ruby. I participated in the Django Girls workshop, then I continued learning django. Although I never worked with ruby on rails, knowledge of django and other MVC frameworks helps me to better understand how ROR works.

I participated in game development in small independent teams, but now I took a break. In those teams I worked with web technologies too. Sometimes I made music and animations in those projects.

All of my listed activities were unpaid, so I hope I can apply to an internship, study at the college, get more experience and finally get a  job. From this September I am pursuing B. S. in C. S. in the University of the People. Until the Outreachy I never contribute to FOSS because I was afraid that my skills are not so good, but now I can already do something for splendid projects, and I'm so glad of it!

If I made some grammar mistakes, I'm sorry, I'm not a native speaker.

Wish you all the best!

**Comment by jywarren** [comment]
Hello! Welcome, @narnt -- we're glad to meet you! I've made a first-time issue for you here, if you're interested!

https://github.com/publiclab/plots2/issues/3838

Django and Rails, and Python and Ruby, are quite closely related, so I agree, you should have a relatively smooth transition. 

We're here to help and very glad to have your assistance as well. Thanks for introducing yourself!

**Comment by hydro-m** [comment]
Hey! I'm another Maria from Russia)
Can you please make an issue for me too?

**Comment by jywarren** [comment]
Hi, @hydro-m -- sure! See https://github.com/publiclab/plots2/issues/3857

**Comment by jywarren** [comment]
Hey everyone -- I heard via the Outreachy mentors list that some applicants are having trouble getting a response from mentors. I want to ask if anyone is "stuck" or looking for an issue, and we will do our best! 

Please forgive us if we've missed something, and remind us in this issue if you have a pull request awaiting review. We've been a bit hard-pressed with the Google Code-In program running, and as you may be able to tell, there are a LOT of contributions coming in fast! But we will do our best to reply quickly if you can chime in here. 

Thanks, everyone! It's great to see all the energy!

**Comment by SidharthBansal** [comment]
Hi @narnt, I am excited to see you here.
Django and ROR as Jeff said are really close.

You can try out a few projects at the Odin project. I learnt ROR from
there. Its really an easy language.
Best of luck

**Comment by narnt** [comment]
@jywarren thank you for the issue for me! I already made one PR, thank you for merging it, can I still work on that fto issue?

**Comment by narnt** [comment]
What are official proposals? In the Public Lab website, I have read that I have to post official proposals on the Outreachy website, but I can't find it on the Outreachy website.

**Comment by amychan331** [comment]
@jywarren My pull request is waiting for review at https://github.com/publiclab/plots2/pull/3720. Thank you!

**Comment by love-opensource** [comment]
Hi all! 
I'm new from Outreachy and I'll like to make my first contribution.
All the first-timers issues are claimed.
Thanks for creating one for me.
Thanks.

**Comment by bansalsidharthcode** [comment]
Nice to hear from you @love-opensource.Can you please introduce yourself?
We will love to hear about you.

**Comment by love-opensource** [comment]
Hi @bansalsidharthcode and everyone. I am Abdellah Ramadan from Cameroon. I'm quite new to open source. I applied to do internship with Outreachy and this is the first of its kind as I am concerned. I have read all the welcoming messages and I'm very excited to work and learn from the community.
I am a student of Software engineering in Saint Monica University Institute in Cameroon.
I will like to make my first contribution here so as to be part of the internship and I wish that you create a first-timer issue for me.

Thanks.

**Comment by SidharthBansal** [comment]
> What are official proposals? In the Public Lab website, I have read that
I have to post official proposals on the Outreachy website, but I can't
find it on the Outreachy website.

@narnt  you can checkout the official proposals for past gsoc and
rgsoc program here.
https://publiclab.org/tag/gsoc
You need to make one on publiclab.org and also submit in on the
Outreachy website before the deadline

**Comment by romanrodriguez** [comment]
Hello everyone, I guess it is better late than never. I joined Outreachy late in the process, and this project really caught my eye. This is such a nice community!

I have some experience in JS, Python, and the front-end. I've toyed around with Ruby and C. I am also really into good documentation. I look forward to learning more about Public Lab and getting more involved.

**Comment by SidharthBansal** [comment]
Sorry. I was reading on mobile and accidentally clicked on the close issue button. Reopened it.

**Comment by jywarren** [comment]
Hi, folks -- as you may know, we've accepted Cess to be our Outreachy intern this session (congratulations, Cess!). I wanted to write to thank everyone for their incredible work over the past several weeks. If we had the capacity, we would have loved to accept more people. Our community is growing and perhaps this will be possible in the future.

I want to wish everyone the best whether you were accepted for another project, or if you have other projects of your own coming up in your life, and to make sure you all know you are always welcome here in our community. You're all amazing and I'm confident you all have amazing things in store in the future!!

THANK YOU for everything!!! 🎉 ❤️
